Source/FeatureRegistry.swift

```diff
@@ -132,6 +132,39 @@ extension LabeledGroupItem { /* Collection Support */
             }
         }
     }
+
+
+}
+
+internal extension Collection where Element == LabeledItem {
+
+
+    /// This function recursivly traverses through the elements in it's array and returns an array of transformed objects.
+    ///
+    /// - Parameters:
+    ///   - groupItems: Array of `LabelGroupItems` types that keeps track of the parent group items of a feature.
+    ///   - resultBuilder: Configuration block for creating the return type at the end of the tree
+    ///   - filter: Includes the element if the given predicate is satisfied
+    func depthFirstMap<T>(groupItems: [LabeledGroupItem] = [],
+        resultBuilder:(([LabeledGroupItem], LabeledFeatureItem) -> T),
+        filter: ((LabeledFeatureItem) -> Bool) = {_ in true }) -> [T] {
+
+        return self.flatMap { labeledItem -> [T] in
+            switch labeledItem {
+            case let featureGroup as LabeledGroupItem:
+                // Traverse into the group and add the group the nextGroupItems
+                var nextGroupItems = groupItems
+                nextGroupItems.append(featureGroup)
+                return featureGroup.depthFirstMap(groupItems: nextGroupItems, resultBuilder: resultBuilder, filter: filter)
+            case let feature as LabeledFeatureItem where filter(feature):
+                // For single items, use the result builder to create the required type
+                return [resultBuilder(groupItems, feature)]
+            default:
+                // For any custom subclass which we don't know about, bail.
+                return []
+            }
+        }
+    }
```

Comment:
This might be a bit much for this feature but we do end up with a flexible way of going through the feature and creating a mapped array.